DB/src/main/java/io/deephaven/db/v2/locations/parquet/topage/ToStringPage.java

```diff
@@ -6,18 +6,26 @@
 import org.apache.parquet.io.api.Binary;
 import org.jetbrains.annotations.NotNull;
 
+import java.lang.reflect.Array;
+
 public class ToStringPage<ATTR extends Attributes.Any> implements ToPage<ATTR, String[]> {
 
-    private static final ToStringPage INSTANCE = new ToStringPage();
+    static final ToStringPage INSTANCE = new ToStringPage();
 
     public static <ATTR extends Attributes.Any>
-    ToPage<ATTR, String[]> create(Class<?> nativeType, org.apache.parquet.column.Dictionary dictionary) {
+    ToPage<ATTR, String[]> create(final Class<?> nativeType, final org.apache.parquet.column.Dictionary dictionary) {
         if (nativeType == null || String.class.equals(nativeType)) {
             //noinspection unchecked
             return dictionary == null ? INSTANCE :
-                    new ToPageWithDictionary<>(String.class, new Dictionary<>( i ->
-                            dictionary.decodeToBinary(i).toStringUsingUTF8(),
-                            dictionary.getMaxId()+1));
+                    new ToPageWithDictionary<>(
+                            String.class,
+                            new Dictionary<>( i -> dictionary.decodeToBinary(i).toStringUsingUTF8(), dictionary.getMaxId()+1),
+                            (final Object result) -> {
+                                final Binary[] from = (Binary[]) result;
+                                final String[] to = (String[]) Array.newInstance(nativeType, from.length);
```

Comment:
`nativeType` always has to be `null` or `String.class`. If it's `null`, this line is going to break, which you don't really want, so you'd rather write `Array.newInstance(String.class, from.length)` or `Array.newInstance(nativeType(), from.length)`, but that simplifies to `new String[from.length]` which doesn't use reflection and is just better in every way, so do that.

tl;dr:
`final String[] to = new String[from.length];`